sherpa/astro/data.py

```diff
@@ -4206,7 +4170,8 @@ def unsubtract(self):
 
 class DataIMG(Data2D):
     "Image data set, including functions for coordinate transformations"
-    _fields = Data2D._fields + ("sky", "eqpos", "coord", "header")
+    _fields = Data2D._fields
```

Comment:
I don't understand your comment. This line is not changed in my PR - perhaps it should be and we should add the sky/.. values to `_extra_fields` instead since they aren't "column data" - but there is a difference here in the `Data2D` and `DataIMG` cases. If I use the current main branch I get

```
>>> from sherpa.data import Data2D
>>> from sherpa.astro.data import DataIMG
>>> Data2D(None, None, None, None)._fields
('name', 'x0', 'x1', 'y', 'shape', 'staterror', 'syserror')
>>> DataIMG(None, None, None, None)._fields
('name', 'x0', 'x1', 'y', 'shape', 'staterror', 'syserror', 'sky', 'eqpos', 'coord', 'header')
```

and we do want these to be different:

```
>>> print(Data2D(None, None, None, None))
name      = None
x0        = None
x1        = None
y         = None
shape     = None
staterror = None
syserror  = None
>>> print(DataIMG(None, None, None, None))
name      = None
x0        = None
x1        = None
y         = None
shape     = None
staterror = None
syserror  = None
sky       = None
eqpos     = None
coord     = logical
```

Reply:
That's strange. Github displays this line as changed for me. Screen shot:
![Screen Shot 2021-11-23 at 10 15 33 AM](https://user-images.githubusercontent.com/498688/143051139-2dcfce47-ecda-44d7-bbf9-2d4bf7a780d5.png)
And the new version seems to be just assigning what's in the base class anyway. I'll reload the page. Maybe I'm stuck at an old version.

Reply:
Hmmm. I do have `_extra_fields` set (see next line) so maybe it is just an error on my part and we can remove it (we can remove it rather than saying `_fields = Data2D._fields` is this is true).

Reply:
I've removed header field from `DataIMG` and so cleaned up both it and `DataIMGInt`